// plugin-welcome/src/api/models/EntAppointment.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntAppointmentEdges,
    EntAppointmentEdgesFromJSON,
    EntAppointmentEdgesFromJSONTyped,
    EntAppointmentEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntAppointment
 */
export interface EntAppointment {
    /**
     * Datetime holds the value of the "datetime" field.
     * @type {string}
     * @memberof EntAppointment
     */
    datetime?: string;
    /**
     * Detail holds the value of the "detail" field.
     * @type {string}
     * @memberof EntAppointment
     */
    detail?: string;
    /**
     * 
     * @type {EntAppointmentEdges}
     * @memberof EntAppointment
     */
    edges?: EntAppointmentEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntAppointment
     */
    id?: number;
}

export function EntAppointmentFromJSON(json: any): EntAppointment {
    return EntAppointmentFromJSONTyped(json, false);
}

export function EntAppointmentFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntAppointment {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'datetime': !exists(json, 'datetime') ? undefined : json['datetime'],
        'detail': !exists(json, 'detail') ? undefined : json['detail'],
        'edges': !exists(json, 'edges') ? undefined : EntAppointmentEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntAppointmentToJSON(value?: EntAppointment | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'datetime': value.datetime,
        'detail': value.detail,
        'edges': EntAppointmentEdgesToJSON(value.edges),
        'id': value.id,
    };
}
