// plugin-welcome/src/api/models/EntDentist.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDentistEdges,
    EntDentistEdgesFromJSON,
    EntDentistEdgesFromJSONTyped,
    EntDentistEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntDentist
 */
export interface EntDentist {
    /**
     * Age holds the value of the "age" field.
     * @type {number}
     * @memberof EntDentist
     */
    age?: number;
    /**
     * Birthday holds the value of the "birthday" field.
     * @type {string}
     * @memberof EntDentist
     */
    birthday?: string;
    /**
     * Cardid holds the value of the "cardid" field.
     * @type {string}
     * @memberof EntDentist
     */
    cardid?: string;
    /**
     * 
     * @type {EntDentistEdges}
     * @memberof EntDentist
     */
    edges?: EntDentistEdges;
    /**
     * Email holds the value of the "email" field.
     * @type {string}
     * @memberof EntDentist
     */
    email?: string;
    /**
     * Experience holds the value of the "experience" field.
     * @type {string}
     * @memberof EntDentist
     */
    experience?: string;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntDentist
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntDentist
     */
    name?: string;
    /**
     * Password holds the value of the "password" field.
     * @type {string}
     * @memberof EntDentist
     */
    password?: string;
    /**
     * Tel holds the value of the "tel" field.
     * @type {string}
     * @memberof EntDentist
     */
    tel?: string;
}

export function EntDentistFromJSON(json: any): EntDentist {
    return EntDentistFromJSONTyped(json, false);
}

export function EntDentistFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntDentist {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'age': !exists(json, 'age') ? undefined : json['age'],
        'birthday': !exists(json, 'birthday') ? undefined : json['birthday'],
        'cardid': !exists(json, 'cardid') ? undefined : json['cardid'],
        'edges': !exists(json, 'edges') ? undefined : EntDentistEdgesFromJSON(json['edges']),
        'email': !exists(json, 'email') ? undefined : json['email'],
        'experience': !exists(json, 'experience') ? undefined : json['experience'],
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'password': !exists(json, 'password') ? undefined : json['password'],
        'tel': !exists(json, 'tel') ? undefined : json['tel'],
    };
}

export function EntDentistToJSON(value?: EntDentist | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'age': value.age,
        'birthday': value.birthday,
        'cardid': value.cardid,
        'edges': EntDentistEdgesToJSON(value.edges),
        'email': value.email,
        'experience': value.experience,
        'id': value.id,
        'name': value.name,
        'password': value.password,
        'tel': value.tel,
    };
}
