// plugin-welcome/src/api/models/EntDentalExpense.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDentalExpenseEdges,
    EntDentalExpenseEdgesFromJSON,
    EntDentalExpenseEdgesFromJSONTyped,
    EntDentalExpenseEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntDentalExpense
 */
export interface EntDentalExpense {
    /**
     * AddedTime holds the value of the "added_time" field.
     * @type {string}
     * @memberof EntDentalExpense
     */
    addedTime?: string;
    /**
     * 
     * @type {EntDentalExpenseEdges}
     * @memberof EntDentalExpense
     */
    edges?: EntDentalExpenseEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntDentalExpense
     */
    id?: number;
}

export function EntDentalExpenseFromJSON(json: any): EntDentalExpense {
    return EntDentalExpenseFromJSONTyped(json, false);
}

export function EntDentalExpenseFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntDentalExpense {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'addedTime': !exists(json, 'added_time') ? undefined : json['added_time'],
        'edges': !exists(json, 'edges') ? undefined : EntDentalExpenseEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntDentalExpenseToJSON(value?: EntDentalExpense | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'added_time': value.addedTime,
        'edges': EntDentalExpenseEdgesToJSON(value.edges),
        'id': value.id,
    };
}
