// plugin-welcome/src/api/models/EntDisease.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDiseaseEdges,
    EntDiseaseEdgesFromJSON,
    EntDiseaseEdgesFromJSONTyped,
    EntDiseaseEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntDisease
 */
export interface EntDisease {
    /**
     * 
     * @type {EntDiseaseEdges}
     * @memberof EntDisease
     */
    edges?: EntDiseaseEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntDisease
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntDisease
     */
    name?: string;
}

export function EntDiseaseFromJSON(json: any): EntDisease {
    return EntDiseaseFromJSONTyped(json, false);
}

export function EntDiseaseFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntDisease {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntDiseaseEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function EntDiseaseToJSON(value?: EntDisease | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntDiseaseEdgesToJSON(value.edges),
        'id': value.id,
        'name': value.name,
    };
}
