// plugin-welcome/src/api/models/EntMedicalCareEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntMedicalfile,
    EntMedicalfileFromJSON,
    EntMedicalfileFromJSONTyped,
    EntMedicalfileToJSON,
    EntPatient,
    EntPatientFromJSON,
    EntPatientFromJSONTyped,
    EntPatientToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntMedicalCareEdges
 */
export interface EntMedicalCareEdges {
    /**
     * Medicalfiles holds the value of the medicalfiles edge.
     * @type {Array<EntMedicalfile>}
     * @memberof EntMedicalCareEdges
     */
    medicalfiles?: Array<EntMedicalfile>;
    /**
     * Patients holds the value of the patients edge.
     * @type {Array<EntPatient>}
     * @memberof EntMedicalCareEdges
     */
    patients?: Array<EntPatient>;
}

export function EntMedicalCareEdgesFromJSON(json: any): EntMedicalCareEdges {
    return EntMedicalCareEdgesFromJSONTyped(json, false);
}

export function EntMedicalCareEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntMedicalCareEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'medicalfiles': !exists(json, 'medicalfiles') ? undefined : ((json['medicalfiles'] as Array<any>).map(EntMedicalfileFromJSON)),
        'patients': !exists(json, 'patients') ? undefined : ((json['patients'] as Array<any>).map(EntPatientFromJSON)),
    };
}

export function EntMedicalCareEdgesToJSON(value?: EntMedicalCareEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'medicalfiles': value.medicalfiles === undefined ? undefined : ((value.medicalfiles as Array<any>).map(EntMedicalfileToJSON)),
        'patients': value.patients === undefined ? undefined : ((value.patients as Array<any>).map(EntPatientToJSON)),
    };
}
