// plugin-welcome/src/api/models/EntMedicalCare.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntMedicalCareEdges,
    EntMedicalCareEdgesFromJSON,
    EntMedicalCareEdgesFromJSONTyped,
    EntMedicalCareEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntMedicalCare
 */
export interface EntMedicalCare {
    /**
     * 
     * @type {EntMedicalCareEdges}
     * @memberof EntMedicalCare
     */
    edges?: EntMedicalCareEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntMedicalCare
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntMedicalCare
     */
    name?: string;
}

export function EntMedicalCareFromJSON(json: any): EntMedicalCare {
    return EntMedicalCareFromJSONTyped(json, false);
}

export function EntMedicalCareFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntMedicalCare {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntMedicalCareEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function EntMedicalCareToJSON(value?: EntMedicalCare | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntMedicalCareEdgesToJSON(value.edges),
        'id': value.id,
        'name': value.name,
    };
}
