// plugin-welcome/src/api/models/EntMedicalfile.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntMedicalfileEdges,
    EntMedicalfileEdgesFromJSON,
    EntMedicalfileEdgesFromJSONTyped,
    EntMedicalfileEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntMedicalfile
 */
export interface EntMedicalfile {
    /**
     * AddedTime holds the value of the "added_time" field.
     * @type {string}
     * @memberof EntMedicalfile
     */
    addedTime?: string;
    /**
     * Detail holds the value of the "detail" field.
     * @type {string}
     * @memberof EntMedicalfile
     */
    detail?: string;
    /**
     * 
     * @type {EntMedicalfileEdges}
     * @memberof EntMedicalfile
     */
    edges?: EntMedicalfileEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntMedicalfile
     */
    id?: number;
}

export function EntMedicalfileFromJSON(json: any): EntMedicalfile {
    return EntMedicalfileFromJSONTyped(json, false);
}

export function EntMedicalfileFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntMedicalfile {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'addedTime': !exists(json, 'added_time') ? undefined : json['added_time'],
        'detail': !exists(json, 'detail') ? undefined : json['detail'],
        'edges': !exists(json, 'edges') ? undefined : EntMedicalfileEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntMedicalfileToJSON(value?: EntMedicalfile | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'added_time': value.addedTime,
        'detail': value.detail,
        'edges': EntMedicalfileEdgesToJSON(value.edges),
        'id': value.id,
    };
}
