// plugin-welcome/src/api/models/EntNurse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntNurseEdges,
    EntNurseEdgesFromJSON,
    EntNurseEdgesFromJSONTyped,
    EntNurseEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntNurse
 */
export interface EntNurse {
    /**
     * 
     * @type {EntNurseEdges}
     * @memberof EntNurse
     */
    edges?: EntNurseEdges;
    /**
     * Email holds the value of the "email" field.
     * @type {string}
     * @memberof EntNurse
     */
    email?: string;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntNurse
     */
    id?: number;
    /**
     * NurseAge holds the value of the "nurse_age" field.
     * @type {number}
     * @memberof EntNurse
     */
    nurseAge?: number;
    /**
     * NurseName holds the value of the "nurse_name" field.
     * @type {string}
     * @memberof EntNurse
     */
    nurseName?: string;
    /**
     * Password holds the value of the "password" field.
     * @type {string}
     * @memberof EntNurse
     */
    password?: string;
}

export function EntNurseFromJSON(json: any): EntNurse {
    return EntNurseFromJSONTyped(json, false);
}

export function EntNurseFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntNurse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntNurseEdgesFromJSON(json['edges']),
        'email': !exists(json, 'email') ? undefined : json['email'],
        'id': !exists(json, 'id') ? undefined : json['id'],
        'nurseAge': !exists(json, 'nurse_age') ? undefined : json['nurse_age'],
        'nurseName': !exists(json, 'nurse_name') ? undefined : json['nurse_name'],
        'password': !exists(json, 'password') ? undefined : json['password'],
    };
}

export function EntNurseToJSON(value?: EntNurse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntNurseEdgesToJSON(value.edges),
        'email': value.email,
        'id': value.id,
        'nurse_age': value.nurseAge,
        'nurse_name': value.nurseName,
        'password': value.password,
    };
}
