// plugin-welcome/src/api/models/EntPriceType.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntPriceTypeEdges,
    EntPriceTypeEdgesFromJSON,
    EntPriceTypeEdgesFromJSONTyped,
    EntPriceTypeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPriceType
 */
export interface EntPriceType {
    /**
     * 
     * @type {EntPriceTypeEdges}
     * @memberof EntPriceType
     */
    edges?: EntPriceTypeEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntPriceType
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntPriceType
     */
    name?: string;
}

export function EntPriceTypeFromJSON(json: any): EntPriceType {
    return EntPriceTypeFromJSONTyped(json, false);
}

export function EntPriceTypeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPriceType {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntPriceTypeEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function EntPriceTypeToJSON(value?: EntPriceType | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntPriceTypeEdgesToJSON(value.edges),
        'id': value.id,
        'name': value.name,
    };
}
