// plugin-welcome/src/api/models/EntQueue.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntQueueEdges,
    EntQueueEdgesFromJSON,
    EntQueueEdgesFromJSONTyped,
    EntQueueEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntQueue
 */
export interface EntQueue {
    /**
     * Dental holds the value of the "dental" field.
     * @type {string}
     * @memberof EntQueue
     */
    dental?: string;
    /**
     * 
     * @type {EntQueueEdges}
     * @memberof EntQueue
     */
    edges?: EntQueueEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntQueue
     */
    id?: number;
    /**
     * QueueTime holds the value of the "queue_time" field.
     * @type {string}
     * @memberof EntQueue
     */
    queueTime?: string;
}

export function EntQueueFromJSON(json: any): EntQueue {
    return EntQueueFromJSONTyped(json, false);
}

export function EntQueueFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntQueue {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'dental': !exists(json, 'dental') ? undefined : json['dental'],
        'edges': !exists(json, 'edges') ? undefined : EntQueueEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'queueTime': !exists(json, 'queue_time') ? undefined : json['queue_time'],
    };
}

export function EntQueueToJSON(value?: EntQueue | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'dental': value.dental,
        'edges': EntQueueEdgesToJSON(value.edges),
        'id': value.id,
        'queue_time': value.queueTime,
    };
}
