// plugin-welcome/src/api/models/EntQueueEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDentist,
    EntDentistFromJSON,
    EntDentistFromJSONTyped,
    EntDentistToJSON,
    EntNurse,
    EntNurseFromJSON,
    EntNurseFromJSONTyped,
    EntNurseToJSON,
    EntPatient,
    EntPatientFromJSON,
    EntPatientFromJSONTyped,
    EntPatientToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntQueueEdges
 */
export interface EntQueueEdges {
    /**
     * 
     * @type {EntDentist}
     * @memberof EntQueueEdges
     */
    dentist?: EntDentist;
    /**
     * 
     * @type {EntNurse}
     * @memberof EntQueueEdges
     */
    nurse?: EntNurse;
    /**
     * 
     * @type {EntPatient}
     * @memberof EntQueueEdges
     */
    patient?: EntPatient;
}

export function EntQueueEdgesFromJSON(json: any): EntQueueEdges {
    return EntQueueEdgesFromJSONTyped(json, false);
}

export function EntQueueEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntQueueEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'dentist': !exists(json, 'dentist') ? undefined : EntDentistFromJSON(json['dentist']),
        'nurse': !exists(json, 'nurse') ? undefined : EntNurseFromJSON(json['nurse']),
        'patient': !exists(json, 'patient') ? undefined : EntPatientFromJSON(json['patient']),
    };
}

export function EntQueueEdgesToJSON(value?: EntQueueEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'dentist': EntDentistToJSON(value.dentist),
        'nurse': EntNurseToJSON(value.nurse),
        'patient': EntPatientToJSON(value.patient),
    };
}
