// plugin-welcome/src/api/models/EntRoomEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntAppointment,
    EntAppointmentFromJSON,
    EntAppointmentFromJSONTyped,
    EntAppointmentToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntRoomEdges
 */
export interface EntRoomEdges {
    /**
     * Appointment holds the value of the appointment edge.
     * @type {Array<EntAppointment>}
     * @memberof EntRoomEdges
     */
    appointment?: Array<EntAppointment>;
}

export function EntRoomEdgesFromJSON(json: any): EntRoomEdges {
    return EntRoomEdgesFromJSONTyped(json, false);
}

export function EntRoomEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntRoomEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'appointment': !exists(json, 'appointment') ? undefined : ((json['appointment'] as Array<any>).map(EntAppointmentFromJSON)),
    };
}

export function EntRoomEdgesToJSON(value?: EntRoomEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'appointment': value.appointment === undefined ? undefined : ((value.appointment as Array<any>).map(EntAppointmentToJSON)),
    };
}
